### PR TITLE
Improve integration-aliases docs.

### PR DESCRIPTION
**Pull request by mattclay**
##### SUMMARY

Improve integration-aliases docs.

##### ISSUE TYPE

Docs Pull Request

##### COMPONENT NAME

integration-aliases sanity test docs

##### ANSIBLE VERSION

```
ansible 2.6.0 (testing-docs 805556dad5) last updated 2018/04/20 09:46:07 (GMT -700)
  config file = None
  configured module search path = [u'/Users/mclay/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/mclay/code/mattclay/ansible/lib/ansible
  executable location = /Users/mclay/code/mattclay/ansible/bin/ansible
  python version = 2.7.14 (default, Mar 22 2018, 11:39:16) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

### CODE REVIEW

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst [hunk 1]
```diff
@@ -1,36 +1,112 @@
 Sanity Tests » integration-aliases
 ==================================
 
-Each integration test must have an ``aliases`` file to control test execution.
+Integration tests are executed by ``ansible-test`` and reside in directories under ``test/integration/targets/``.
+Each test must have an ``aliases`` file to control test execution.
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst]: s/must/MUST/

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst [hunk 2]
```diff
@@ -1,36 +1,112 @@
...
+Aliases are explained in the following sections. Each alias must be on a separate line in an ``aliases`` file.
 
-In some cases tests requiring external resources can be run as a part of CI.
-This is often true when those resources can be provided by a docker container.
+Groups
+------
 
-However, if you think that the tests should be able to be supported by CI, please discuss test
-organization with @mattclay or @gundalow on GitHub or #ansible-devel on IRC.
+Tests must be configured to run in a specific group. This is done by adding the appropriate group to the ``aliases`` file.
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst]: > run in a specific group

Run in exactly one?
Run in at least one?

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst [hunk 3]
```diff
@@ -1,36 +1,112 @@
...
+- ``posix/ci/cloud/group3/azure``
+- ``posix/ci/cloud/group4/aws``
 
-The CI groups are used to balance tests across multiple jobs to minimize test run time.
-Using the relevant ``group1`` entry is fine in most cases. Groups can be changed later to redistribute tests.
+Groups are used to balance tests across multiple CI jobs to minimize test run time.
+They also improve efficiency by keeping tests with similar requirements running together.
 
-Aliases can also be used to express test requirements:
+When selecting a group for a new test, use the same group as existing tests similar to the one being added.
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst:24]: Worth saying anything about which group`N` to use?
ie. just pick a random one?

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst [hunk 4]
```diff
@@ -1,36 +1,112 @@
...
+
+- ``skip/freebsd`` - Skip tests on FreeBSD.
+- ``skip/osx`` - Skip tests on macOS / OS X.
+- ``skip/rhel`` - Skip tests on RHEL.
+
+Aliases can be used to skip Python major versions:
+
+- ``skip/python2`` - Skip tests on Python 2.x.
+- ``skip/python3`` - Skip tests on Python 3.x.
+
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst]: Note to say if you require finer grained skipping use `when: ansible_distribution in ('Ubuntu')`

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst [hunk 5]
```diff
@@ -1,36 +1,112 @@
...
-- ``skip/python3``
+- ``needs/privileged`` - Requires ``--docker-privileged`` when running tests with ``--docker``.
+- ``needs/root`` - Requires running tests as ``root`` or with ``--docker``.
+- ``needs/ssh`` - Requires SSH connections to localhost (or the test container with ``--docker``) without a password.
 
-Take a look at existing ``aliases`` files to see what aliases are available and how they're used.
+Skipping
+--------
+
+Aliases can be used to skip platforms:
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst]: Make it clear if these are the only platforms we support skipping on

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst [hunk 6]
```diff
@@ -1,36 +1,112 @@
...
+Tests which fail sometimes should be marked with the ``unstable`` alias until the instability has been fixed.
+These tests will continue to run for pull requests which modify the test or the module under test.
+
+This avoids unnecessary test failures for other pull requests, as well as tests on merge runs and nightly CI jobs.
+
+There are two ways to run unstable tests manually:
+
+- Use the ``--allow-unstable`` option for ``ansible-test``
+- Prefix the test name with ``unstable/`` when passing it to ``ansible-test``.
+
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst:69]: When should people add the unstable alias?
Should a GitHub issue be raised?

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst [hunk 7]
```diff
@@ -1,36 +1,112 @@
...
+Tests which always fail should be marked with the ``disabled`` alias until they can be fixed.
+
+Disabled tests are automatically skipped.
+
+There are two ways to run disabled tests manually:
+
+- Use the ``--allow-disabled`` option for ``ansible-test``
+- Prefix the test name with ``disabled/`` when passing it to ``ansible-test``.
+
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst:84]: Same questions as unstable

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst [hunk 8]
```diff
@@ -1,36 +1,112 @@
...
+Tests which cannot be run in CI should be marked with the ``unsupported`` alias.
+Most tests can be supported through the use of simulators and/or cloud plugins.
+
+However, if that is not possible then marking a test as unsupported will prevent it from running in CI.
+
+There are two ways to run unsupported tests manually:
+
+* Use the ``--allow-unsupported`` option for ``ansible-test``
+* Prefix the test name with ``unsupported/`` when passing it to ``ansible-test``.
+
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst:100]: Same questions as unstable

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst [hunk 9]
```diff
@@ -1,36 +1,112 @@
...
+Untested
+--------
+
+Every module and plugin should have integration tests, even if the tests cannot be run in CI.
+
+Questions
+---------
+
+For questions about integration tests reach out to @mattclay or @gundalow on GitHub or #ansible-devel on IRC.
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst]:
```
`#ansible-devel`
```

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
Doc updates look OK mechnically; OK to merge once @gundalow's tech review comments are addressed. Thank @mattclay!

**Review by dharmabumstead — APPROVED** [review]

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst (hunk 3 above)
Comment (mattclay) [reply docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst:24]: I added a sentence below to address this.

**Review by gundalow — APPROVED** [review]
Looks good, thanks

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst (hunk 6 above)
Comment (gundalow) [reply docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst:69]: Thoughts?

Comment (mattclay) [reply docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst:69]: This should be addressed by the additional verbiage below.

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst (hunk 7 above)
Comment (mattclay) [reply docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst:84]: This should be addressed by the additional verbiage below.

File: docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst (hunk 8 above)
Comment (mattclay) [reply docs/docsite/rst/dev_guide/testing/sanity/integration-aliases.rst:100]: This should be addressed by the additional verbiage below.

**Review by mattclay — COMMENTED** [review]
I believe I've addressed all review feedback with my latest commit.

**Review by gundalow — APPROVED** [review]

**Pull request merged**